mypy/newsemanal/semanal.py

```diff
@@ -3797,8 +3759,12 @@ def qualified_name(self, n: str) -> str:
             base = self.cur_mod_id
         return base + '.' + n
 
-    def enter(self) -> None:
-        self.locals.append(SymbolTable())
+    def enter(self, function: Optional[FuncItem] = None) -> None:
+        if function:
```

Comment:
Add docstring and describe the argument. Also discuss when it's okay to leave it out.